apps/condo/domains/common/hooks/useInputWithCounter.tsx

```diff
@@ -0,0 +1,31 @@
+import { InputProps, Typography } from 'antd'
+import React, { useCallback, useState } from 'react'
+import { TextAreaProps } from 'antd/es/input'
+import get from 'lodash/get'
+
+type AntdInputProps = InputProps | TextAreaProps
+
+export const useInputWithCounter = (InputComponent, maxLength) => {
+    const [textLength, setTextLength] = useState<number>(0)
+
+    const InputWithCounter: React.FC<AntdInputProps> = useCallback((inputProps) => (
+        <InputComponent
+            maxLength={maxLength}
+            {...inputProps}
+            onChange={(e) => {
+                const valueLength = get(e, ['target', 'value', 'length'])
+                setTextLength(valueLength)
+
+                inputProps.onChange && inputProps.onChange(e)
```

Comment:
please don't use ternary operator to call functions.
`if (isFunction(inputProps.onChange)) inputProps.onChange(e)`